src/app/public/testing/modal-fixture.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  ComponentFixture
+} from '@angular/core/testing';
+
+/**
+ * Allows interaction with a SKY UX modal component.
+ */
+export class SkyModalFixture {
+
+  private modalElement: HTMLElement;
+
+  constructor(
+    private fixture: ComponentFixture<any>,
+    skyTestId: string
+  ) {
+    this.modalElement = document.querySelector('sky-modal[data-sky-id="' + skyTestId + '"]');
+
+    if (!this.modalElement) {
+      throw new Error(`No element was found with a \`data-sky-id\` value of "${skyTestId}".`);
+    }
+  }
+
+  /**
+   * The modal component's ARIA describedby attribute.
+   */
+  public get ariaDescribedBy(): string {
+    const modalElement = this.getModalDiaglogElement();
+    return modalElement.getAttribute('aria-describedby') || undefined;
+  }
+
+  /**
+   * The modal component's ARIA labelledby attribute.
+   */
+  public get ariaLabelledBy(): string {
+    const modalElement = this.getModalDiaglogElement();
+    return modalElement.getAttribute('aria-labelledby') || undefined;
+  }
+
+  /**
+   * The modal component's role attribute.
+   */
+  public get ariaRole(): string {
+    const modalElement = this.getModalDiaglogElement();
+    return modalElement.getAttribute('role') || undefined;
+  }
+
+  /**
+   * Whether or not the modal is a full page modal
```

Comment:
Minor note - some of your comments don't have punctuation while others do.

Reply:
Added punctuation throughout. Let me know if I should have removed all of them :) It appeared that we normally have them in our documentation.